packages/wrangler/src/config/config-helpers.ts

```diff
@@ -1,35 +1,127 @@
-import path from "path";
+import fs from "node:fs";
+import path from "node:path";
 import { findUpSync } from "find-up";
+import dedent from "ts-dedent";
+import { UserError } from "../errors";
+import { logger } from "../logger";
+import { parseJSONC, readFileSync } from "../parse";
+
+export type ResolveConfigPathOptions = {
+	useRedirect?: boolean;
+};
 
 /**
  * Resolve the path to the configuration file, given the `config` and `script` optional command line arguments.
  * `config` takes precedence, then `script`, then we just use the cwd.
  */
-export function resolveWranglerConfigPath({
-	config,
-	script,
-}: {
-	config?: string;
-	script?: string;
-}): string | undefined {
+export function resolveWranglerConfigPath(
+	{
+		config,
+		script,
+	}: {
+		config?: string;
+		script?: string;
+	},
+	options: { useRedirect?: boolean }
+): string | undefined {
 	if (config !== undefined) {
 		return config;
 	}
 
 	const leafPath = script !== undefined ? path.dirname(script) : process.cwd();
-	return findWranglerConfig(leafPath);
+
+	return findWranglerConfig(leafPath, options);
 }
 
 /**
  * Find the wrangler config file by searching up the file-system
  * from the current working directory.
  */
 export function findWranglerConfig(
-	referencePath: string = process.cwd()
+	referencePath: string = process.cwd(),
+	{ useRedirect = false } = {}
 ): string | undefined {
-	return (
+	const userConfigPath =
 		findUpSync(`wrangler.json`, { cwd: referencePath }) ??
 		findUpSync(`wrangler.jsonc`, { cwd: referencePath }) ??
-		findUpSync(`wrangler.toml`, { cwd: referencePath })
-	);
+		findUpSync(`wrangler.toml`, { cwd: referencePath });
+
+	return useRedirect
+		? findRedirectedWranglerConfig(referencePath, userConfigPath)
+		: userConfigPath;
+}
+
+/**
+ * Check whether there is a config file that indicates
+ * @param cwd
+ * @param userConfigPath
+ * @returns
+ */
+function findRedirectedWranglerConfig(
+	cwd: string,
+	userConfigPath: string | undefined
+) {
+	const PATH_TO_DEPLOY_CONFIG = ".wrangler/deploy/config.json";
+	const deployConfigPath = findUpSync(PATH_TO_DEPLOY_CONFIG, { cwd });
```

Comment:
~Should this be a `findUpSync()`? I almost feel like this should just look directly for that path starting from the dirname of the user config file~

Never mind, I see this is an error further down